File: api/models/account.js
const mongoose = require('mongoose');
const bcrypt = require('bcrypt');

const AccountSchema = new mongoose.Schema({
    email: { type: String, required: true, unique: true },
    password: { type: String, required: true },
    fullName: { type: String, default: "" },
    address: { type: String, default: "" },
    phoneNumber: { type: String, default: "" },
    bod: { type: Date, default: null },
    sex: { type: Number, default: 1 }
});

AccountSchema.pre('save', async function(next) {
    try {
        const account = this;
        if (!account.isModified('password')) return next();

        const salt = await bcrypt.genSalt(10);
        const hashPass = await bcrypt.hash(account.password, salt);

        account.password = hashPass;
        next();
    } catch (err) {
        next(err);
    }
});

AccountSchema.methods.comparePassword = async function(userPassword){
    try {
        const isMatch = await bcrypt.compare(userPassword, this.password)
        return isMatch
    } catch (error) {
        throw error;
    }
}

module.exports = mongoose.model("Account", AccountSchema);

File: admin_vbooks/lib/data/model/account.dart
class Account {
  String? id;
  String? email;
  String? password;
  String? fullName;
  String? address;
  String? phoneNumber;
  String? dob;
  int? sex;

  Account(
      {this.id,
      this.email,
      this.password,
      this.fullName,
      this.address,
      this.phoneNumber,
      this.dob,
      this.sex});

  Account.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    email = json['email'];
    password = json['password'];
    fullName = json['fullName'];
    address = json['address'];
    phoneNumber = json['phoneNumber'];
    dob = json['bod'];
    sex = json['sex'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['_id'] = this.id;
    data['email'] = this.email;
    data['password'] = this.password;
    data['fullName'] = this.fullName;
    data['address'] = this.address;
    data['phoneNumber'] = this.phoneNumber;
    data['bod'] = this.dob;
    data['sex'] = this.sex;
    return data;
  }

  factory Account.fromMap(Map<String, dynamic> map) {
    return Account(
      id: map['_id'].toInt() ?? 0,
      email: map['email']?.toString() ?? '',
      password: map['password']?.toString() ?? '',
      fullName: map['fullName']?.toString() ?? '',
      address: map['address']?.toString() ?? '',
      sex: map['sex']?.toInt() ?? 1
    );
  }
}
